inc/customizer/controls/layouts.php

```diff
@@ -12,7 +12,43 @@ public function enqueue() {
 		$rtl    = is_rtl() ? '-rtl' : '';
 		$suffix = SCRIPT_DEBUG ? '' : '.min';
 
-		wp_enqueue_style( 'primer-layouts', get_template_directory_uri() . "/assets/css/admin/layouts{$rtl}{$suffix}.css", array(), PRIMER_VERSION );
+		wp_enqueue_style(
+			'primer-layouts',
+			get_template_directory_uri() . "/assets/css/admin/layouts{$rtl}{$suffix}.css",
+			array(),
+			PRIMER_VERSION
+		);
+
+		wp_enqueue_script(
+			'primer-layouts',
+			get_template_directory_uri() . "/assets/js/admin/layouts{$suffix}.js",
+			array( 'customize-controls' ),
+			true
+		);
+
+		$layouts = array_fill_keys( array_keys( $this->choices ), 'postMessage' );
+
+		/**
+		 * Those layouts are loaded dynamically and need refresh
+		 */
+		$layouts['three-column-default']  = 'refresh';
+		$layouts['three-column-center']   = 'refresh';
+		$layouts['three-column-reversed'] = 'refresh';
+
+		/**
+		 * Filter layouts transport mechanism in the customizer. Either postMessage or refresh.
+		 *
+		 * @since 1.0.0
+		 *
+		 * @var array
+		 */
+		$layouts = apply_filters( 'primer_layouts_transport', $layouts );
```

Comment:
@jonathanbardo As a best-practice we should type cast filter results. Done in https://github.com/godaddy/wp-primer-theme/commit/4165462e0b03c815bc79cb5fa4b2161d0c68d7e1